### ISSUE TITLE
KeyboardAvoidingView Android white area when keyboard open/close 

### ISSUE THREAD

**Issue by Nazort** [facebook/react-native#32612]
### Description

When using KeyboardAvoidingView on android, whatever the configuration I use for the `behavior` or the `keyboardVerticalOffset` I always have a white area when the keyboard is opening or closing while on iOS everything is moving smoothly.

https://user-images.githubusercontent.com/21194324/142198993-243f9d37-07ae-4b10-bc7d-2357867629fb.mp4

In the video, we can see that when the keyboard opens there is a big white area and when closing sometimes a small white rectangle with some flickering. I'm wondering if this is normal behavior because on iOS it transitions smoothly 

I tried modifying android manifest with  android:windowSoftInputMode="adjustResize" or  `android:windowSoftInputMode="adjustResize"` `android:windowSoftInputMode="adjustPan"` but the problem is still there with both config. 

It would actually be ok if we could change this color from white to something else like the background color of the content (red in my example). But I couldn't find a way.

### Version

0.66	

### Output of `react-native info`

It happens from the documentation snack so I guess this section is not necessary 

### Steps to reproduce

Take the example from the KeyboardAvoidingView of react-native documentation and modify the background color of the container from white to red in order to see the white area when opening/closing the keyboard.

### Snack, code example, screenshot, or link to a repository

https://snack.expo.dev/@flowwwww/keyboardavoidingview

**Comment by oguzturker8** [comment]
Hi, have you tried to re-build after changing `android:windowSoftInputMode:` ?
When `behavior='height'` on Android, following code shows we return `initialHeight - bottomHeight`.
When keyboard is opening the remaining bottomHeight px area shows up as white blank area.

in KeyboardAvoidingView.js
```js
  switch (behavior) {
      case 'height':
        let heightStyle;
        if (this._frame != null && this.state.bottom > 0) {
          // Note that we only apply a height change when there is keyboard present,
          // i.e. this.state.bottom is greater than 0. If we remove that condition,
          // this.frame.height will never go back to its original value.
          // When height changes, we need to disable flex.
          heightStyle = {
            height: this._initialFrameHeight - bottomHeight,
            flex: 0,
          };
        }
        return (
          <View
            ref={this.viewRef}
            style={StyleSheet.compose(style, heightStyle)}
            onLayout={this._onLayout}
            {...props}>
            {children}
          </View>
        );
```

**Comment by kabus202** [comment]
I have the same. Did you find any solution @Nazort ??

**Comment by KubaKubikula** [comment]
same here

**Comment by omri81** [comment]
facing the same issue anyone has a solution?

**Comment by JaLe29** [comment]
same, no solution

**Comment by baesumin** [comment]
+1

**Comment by Polad10** [comment]
+1

**Comment by davram88** [comment]
If you are using expo, try expo-system-ui and set background color. It worked for me!

**Comment by GrantBartlett** [comment]
Fixed by adding `<item name="android:windowBackground">@color/windowBackgroundColor</item>` within the style xml

Found in `android/app/src/main/res/values/styles.xml`
```xml
<style name="AppTheme" parent="Theme.AppCompat.DayNight.NoActionBar">
 ...
<item name="android:windowBackground">@color/windowBackgroundColor</item>
</style>
```

Then create  `android/app/src/main/res/values/colors.xml`

```xml
<resources>
    <color name="windowBackgroundColor">#000000</color>
</resources>
```

**Comment by Nantris** [comment]
This makes the app feel so janky and poorly built no matter how much effort was really put into it, and so far I can't find any solution.

**Comment by iAmHammad261** [comment]
> I fixed this by adding `<item name="android:windowBackground">@color/windowBackgroundColor</item>` within the `styles.xml`
> 
> Add to `android/app/src/main/res/values/styles.xml` so it becomes like:
> 
> ```
> <resources>
> 
>     
>     <style name="AppTheme" parent="Theme.AppCompat.DayNight.NoActionBar">
>         
>         <item name="android:editTextBackground">@drawable/rn_edit_text_material</item>
>         
>         <item name="android:windowBackground">@color/windowBackgroundColor</item>
>     </style>
> 
> </resources>
> ```
> 
> Then create `android/app/src/main/res/values/colors.xml`
> 
> ```
> <resources>
>     <color name="windowBackgroundColor">#000000</color>
> </resources>
> ```
> 
> Can name `windowBackgroundColor` whatever you want just so it matches the colors.xml name.

This Fixed my issue ❤

**Comment by Odyrac** [comment]
+1 no solution find

**Comment by Nantris** [comment]
If you use Expo I'm pretty sure this is the solution: https://docs.expo.dev/versions/latest/sdk/system-ui/#systemuisetbackgroundcolorasynccolor

I assume non-Expo users can figure out what that's doing and leverage it directly.

**Comment by abrahamgalue** [comment]
> If you use Expo I'm pretty sure this is the solution: https://docs.expo.dev/versions/latest/sdk/system-ui/#systemuisetbackgroundcolorasynccolor
> 
> I assume non-Expo users can figure out what that's doing and leverage it directly.

This does indeed allow you to change the background color that appears when the keyboard is opened, but my question now is, how do I make it dynamic? I have a light and dark theme, and if I set a static value when the user switches themes, the user experience will still be poor. Thanks for the alternative.

**Comment by Nantris** [comment]
@abrahamgalue does `SystemUI.setBackgroundColorAsync()` from that link not do it for you? We just call it when the dark/light mode is changed and it works well for us (Expo 52.)

**Comment by abrahamgalue** [comment]
> [@abrahamgalue](https://github.com/abrahamgalue) does `SystemUI.setBackgroundColorAsync()` from that link not do it for you? We just call it when the dark/light mode is changed and it works well for us (Expo 52.)

Do you have any examples of how to do this? I'd really appreciate it. I'm new to Expo and would like to fix this visual effect, but I don't know how to implement it.

**Comment by Nantris** [comment]
@abrahamgalue this is probably what you're looking for. You can listen to changes in dark/light and then set your preferred color: https://reactnative.dev/docs/appearance#addchangelistener

**Comment by abrahamgalue** [comment]
The problem persists

**Comment by Nantris** [comment]
@abrahamgalue on what Android version? I think this is an Android problem with React Native and it seems resolved on newer versions of Android.

**Comment by abrahamgalue** [comment]
@Nantris Well, I went back to the Expo Snack and tried Expo Go on my Android 15 device, and I'm still seeing the problem that was reported when the keyboard jumps—there's a white background on a white background

**Comment by Nantris** [comment]
I don't know if Expo Go is a valid test for this.

Try this to set the background color of the native root component (I guess is what this does) and make a custom dev client build and if you still see it then it's definitely still an issue: https://docs.expo.dev/versions/latest/sdk/system-ui/#systemuisetbackgroundcolorasynccolor

**Comment by abrahamgalue** [comment]
@Nantris Yes, that works. Although I can't stop thinking about how a developer who doesn't use Expo, only React Native as a library, would do it.

**Comment by Nantris** [comment]
Glad to hear it!

I think these are the underlying native functions:

https://github.com/expo/expo/blob/27b2a908664cc5521961a65709dcb8c7e03e54ec/packages/expo-system-ui/ios/ExpoSystemUI/ExpoSystemUIModule.swift#L42

https://github.com/expo/expo/blob/27b2a908664cc5521961a65709dcb8c7e03e54ec/packages/expo-system-ui/android/src/main/java/expo/modules/systemui/SystemUIModule.kt#L40
